kuma/users/tests/test_views.py

```diff
@@ -1601,6 +1601,48 @@ def test_delete_user_keep_attributions(
     assert not Key.objects.all().exists()
 
 
+def test_delete_user_no_revisions_but_attachment_revisions_donate(
+    db, user_client, wiki_user, django_user_model
+):
+    """
+    This test is based on the bug report
+    https://github.com/mdn/kuma/issues/6479
+
+    The user didn't have any revisions to confront the legacy of, but there might be
+    other things attached to the user.
+    """
+    other_user = django_user_model.objects.create(
+        username="other", email="other@example.com"
+    )
+    assert not Revision.objects.filter(creator=wiki_user).update(creator=other_user)
+
+    attachment_revision = AttachmentRevision(
+        attachment=Attachment.objects.create(title="test attachment"),
+        file="some/path.ext",
+        mime_type="application/kuma",
+        creator=wiki_user,
+        title="test attachment",
+    )
+    attachment_revision.save()
+    url = reverse("users.user_delete", kwargs={"username": wiki_user.username})
+    response = user_client.post(url, HTTP_HOST=settings.WIKI_HOST)
+    # This means it didn't work! The form rejects.
+    assert response.status_code == 200
+
+    # Ok, let's donate the attachment revisions to "Anonymous"
+    response = user_client.post(
+        url, {"attributions": "donate"}, HTTP_HOST=settings.WIKI_HOST
+    )
+    # This means it didn't work! The form rejects.
```

Comment:
This comment is off, maybe something like this:
```suggestion
    # This means it worked! The user's attributions have been donated to the Anonymous user.
```